demo/angular/src/app/voice_widget.component.ts

```diff
@@ -1,99 +1,85 @@
-import { Component } from "@angular/core"
-import { Subscription } from "rxjs"
+import { Component, OnDestroy } from '@angular/core';
+import { Subscription } from 'rxjs';
 
-import { PorcupineService, PorcupineServiceArgs } from "@picovoice/porcupine-web-angular"
+import { PorcupineService } from '@picovoice/porcupine-angular';
 
-import { DEEP_SKY_BLUE_PPN_64 } from "./porcupine_keywords"
-import { PorcupineKeyword } from "@picovoice/porcupine-web-core"
+import { BuiltInKeyword } from '@picovoice/porcupine-web';
+
+import porcupineParams from '../lib/porcupine_params';
 
 @Component({
-  selector: 'voice-widget',
+  selector: 'app-voice-widget',
   templateUrl: './voice_widget.component.html',
   styleUrls: ['./voice_widget.component.scss']
 })
-export class VoiceWidget {
-  private keywordDetection: Subscription
-  private listeningDetection: Subscription
-  private errorDetection: Subscription
-  private isErrorDetection: Subscription
+export class VoiceWidget implements OnDestroy {
+  private keywordDetection: Subscription;
+  private isLoadedDetection: Subscription;
+  private isListeningDetection: Subscription;
+  private errorDetection: Subscription;
```

Comment:
A bit confusing to call them all `*Detection`. I would go with either `*Subscription` or `*Event`